Fabric.Authorization.Persistence.SqlServer/Stores/SqlServerPermissionStore.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Fabric.Authorization.Domain.Exceptions;
+using Fabric.Authorization.Domain.Models;
+using Fabric.Authorization.Domain.Stores;
+using Fabric.Authorization.Persistence.SqlServer.Mappers;
+using Fabric.Authorization.Persistence.SqlServer.Services;
+using Microsoft.EntityFrameworkCore;
+
+namespace Fabric.Authorization.Persistence.SqlServer.Stores
+{
+    public class SqlServerPermissionStore : IPermissionStore
+    {
+        private readonly IAuthorizationDbContext _authorizationDbContext;
+
+        public SqlServerPermissionStore(IAuthorizationDbContext authorizationDbContext)
+        {
+            _authorizationDbContext = authorizationDbContext;
+        }
+
+        public async Task<Permission> Add(Permission model)
+        {
+            var entity = model.ToEntity();
+
+            _authorizationDbContext.Permissions.Add(entity);
+            await _authorizationDbContext.SaveChangesAsync();
+            return model;
+        }
+
+        public async Task<Permission> Get(Guid id)
+        {
+            var permission = await _authorizationDbContext.Permissions
+                .SingleOrDefaultAsync(p =>
```

Comment:
Do any of the navigation properties need to be included in the query?

Reply:
In this context I don't think so. I think when retrieving groups/roles and users we will pull in the corresponding permissions, but I envisioned this as just a lookup resource to display permissions. Perhaps @michaelvidal24 can comment on this.

Reply:
I think that makes sense, unless/until we need to pull in all the nav properties we shouldn't.